exp/api/v1alpha3/awsmachinepool_webhook.go

```diff
@@ -0,0 +1,91 @@
+package v1alpha3
+
+import (
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var log = logf.Log.WithName("awsmachinepool-resource")
+
+// SetupWebhookWithManager will setup the webhooks for the AWSMachinePool
+func (r *AWSMachinePool) SetupWebhookWithManager(mgr ctrl.Manager) error {
```

Comment:
This will need to be called from the main if the EKS flag is enabled like [we do for the AWSManagedMachinePool](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/main.go#L227)

Reply:
Thanks..added